compiler/rustc_passes/src/loops.rs

```diff
@@ -24,22 +25,38 @@ enum Context {
     Closure(Span),
     AsyncClosure(Span),
     UnlabeledBlock(Span),
+    IfUnlabeledBlock(Span),
     LabeledBlock,
     Constant,
 }
 
-#[derive(Copy, Clone)]
+#[derive(Clone)]
+struct BlockInfo {
+    name: String,
+    spans: Vec<Span>,
+    suggs: Vec<Span>,
+}
+
+#[derive(Clone)]
 struct CheckLoopVisitor<'a, 'tcx> {
     sess: &'a Session,
     tcx: TyCtxt<'tcx>,
-    cx: Context,
+    // Used for diagnostic like when in a `if` block with some `break`s,
+    // we should not suggest adding `'block` label in `if` block,
+    // we can back to outer block and add label there.
```

Comment:
```suggestion
    // Keep track of a stack of contexts, so that suggestions
    // are not made for contexts where it would be incorrect,
    // such as adding a label for an `if`.
    // e.g. `if 'foo: {}` would be incorrect.
```

Reply:
Done. Thank you.